File: src/main/java/me/yoursole/ctf/CTF.java
package me.yoursole.ctf;

import me.yoursole.ctf.Commands.*;
import me.yoursole.ctf.DataFiles.GameData;
import me.yoursole.ctf.DataFiles.GameLoop;
import me.yoursole.ctf.DataFiles.Items.GoblinsSword;
import me.yoursole.ctf.DataFiles.Items.HermesBoots;
import me.yoursole.ctf.DataFiles.Items.ThorsAxe;
import me.yoursole.ctf.DataFiles.Items.TunnelersPickaxe;
import me.yoursole.ctf.DataFiles.Utils;
import me.yoursole.ctf.Events.*;
import org.bukkit.Bukkit;
import org.bukkit.NamespacedKey;
import org.bukkit.inventory.Recipe;
import org.bukkit.plugin.java.JavaPlugin;

import java.util.Objects;

public final class CTF extends JavaPlugin {

    @Override
    public void onEnable() {
        getServer().getPluginManager().registerEvents(new MoveItemEvent(), this);
        getServer().getPluginManager().registerEvents(new DropItemEvent(), this);
        getServer().getPluginManager().registerEvents(new PlayerDeath(), this);
        getServer().getPluginManager().registerEvents(new GetCompass(), this);
        getServer().getPluginManager().registerEvents(new PlaceFlagEvent(), this);
        getServer().getPluginManager().registerEvents(new Join(), this);
        getServer().getPluginManager().registerEvents(new Leave(), this);
        getServer().getPluginManager().registerEvents(new ChatEvent(), this);
        getServer().getPluginManager().registerEvents(new BlockBreak(), this);
        getServer().getPluginManager().registerEvents(new HermesBoots(), this);
        getServer().getPluginManager().registerEvents(new TunnelersPickaxe(), this);
        getServer().getPluginManager().registerEvents(new GoblinsSword(), this);
        getServer().getPluginManager().registerEvents(new ThorsAxe(), this);
        getServer().getPluginManager().registerEvents(new BuiltLimit(), this);
        getServer().getPluginManager().registerEvents(new ItDamagePlayer(), this);
        getServer().getPluginManager().registerEvents(new CraftEvent(), this);
        getServer().getPluginManager().registerEvents(new ItLocationManager(), this);
        getServer().getPluginManager().registerEvents(new RespawnEvent(), this);
        Objects.requireNonNull(this.getCommand("start")).setExecutor(new Start());
        Objects.requireNonNull(this.getCommand("stop")).setExecutor(new Stop());
        Objects.requireNonNull(this.getCommand("scores")).setExecutor(new Scores());
        Objects.requireNonNull(this.getCommand("timer")).setExecutor(new Timer());
        Objects.requireNonNull(this.getCommand("togglechat")).setExecutor(new ToggleChat());
        Objects.requireNonNull(this.getCommand("customitem")).setExecutor(new CustomItem());
        Objects.requireNonNull(this.getCommand("top")).setExecutor(new Top());

        new GameLoop();
    }

    @Override
    public void onDisable() {
        for (NamespacedKey key : GameData.recipeKeys.values()) {
            Bukkit.removeRecipe(key);
        }
        Bukkit.getScheduler().cancelTask(GameLoop.taskId);

        Utils.deleteWorlds();

    }
}

File: src/main/kotlin/me/yoursole/ctf/CTF.kt
package me.yoursole.ctf

import me.yoursole.ctf.commands.*
import me.yoursole.ctf.datafiles.*
import me.yoursole.ctf.datafiles.items.*
import me.yoursole.ctf.events.*
import org.bukkit.Bukkit
import org.bukkit.plugin.java.JavaPlugin

class CTF : JavaPlugin() {

    companion object {
        lateinit var instance: CTF
    }

    init {
        instance = this
    }

    override fun onEnable() {
        server.pluginManager.registerEvents(WorldManager, this)
        server.pluginManager.registerEvents(MoveItemEvent, this)
        server.pluginManager.registerEvents(DropItemEvent, this)
        server.pluginManager.registerEvents(PlayerDeath, this)
        server.pluginManager.registerEvents(GetCompass, this)
        server.pluginManager.registerEvents(PlaceFlagEvent, this)
        server.pluginManager.registerEvents(Join, this)
        server.pluginManager.registerEvents(Leave, this)
        server.pluginManager.registerEvents(ChatEvent, this)
        server.pluginManager.registerEvents(BlockBreak, this)
        server.pluginManager.registerEvents(HermesBoots, this)
        server.pluginManager.registerEvents(TunnelersPickaxe, this)
        server.pluginManager.registerEvents(GoblinsSword, this)
        server.pluginManager.registerEvents(ThorsAxe, this)
        server.pluginManager.registerEvents(BuildLimit, this)
        server.pluginManager.registerEvents(ItDamagePlayer, this)
        server.pluginManager.registerEvents(CraftEvent, this)
        server.pluginManager.registerEvents(ItLocationManager, this)
        server.pluginManager.registerEvents(RespawnEvent, this)
        server.pluginManager.registerEvents(ChunkLoad, this)
        server.pluginManager.registerEvents(EntityDeath, this)
        server.pluginManager.registerEvents(FlagDropper, this)
        server.pluginManager.registerEvents(PyromancersCharm, this)
        server.pluginManager.registerEvents(RecipeManager, this)
        server.pluginManager.registerEvents(MachineGunBow, this)
        server.pluginManager.registerEvents(OreFinder, this)

        getCommand("start")!!.setExecutor(Start)
        getCommand("stop")!!.setExecutor(Stop)
        getCommand("scores")!!.setExecutor(Scores)
        getCommand("timer")!!.setExecutor(Timer)
        getCommand("togglechat")!!.setExecutor(ToggleChat)
        getCommand("customitem")!!.setExecutor(CustomItem)
        getCommand("top")!!.setExecutor(Top)
        getCommand("test")!!.setExecutor(Test)
        GameLoop()

        for (player in Bukkit.getOnlinePlayers()) {
            GameData.scores[player.uniqueId] = 0
            player.discoverRecipes(GameData.recipeKeys.values)
        }
    }

    override fun onDisable() {
        for (key in GameData.recipeKeys.values) {
            Bukkit.removeRecipe(key)
        }
        Bukkit.getScheduler().cancelTask(GameLoop.taskId)
        WorldManager.deleteWorlds()
        for (player in Bukkit.getOnlinePlayers()) {
            player.isGlowing = false
        }
    }
}